model-optimizer/extensions/middle/UnsqueezeTileReshapeBlockToInterpolate.py

```diff
@@ -22,34 +22,58 @@
 from extensions.ops.elementwise import Mul
 from extensions.ops.interpolate import Interpolate
 from mo.front.common.partial_infer.utils import int64_array, float32_array
+from mo.front.tf.graph_utils import create_op_with_const_inputs
 from mo.graph.graph import Graph, rename_nodes
 from mo.middle.replacement import MiddleReplacementPattern
-from mo.ops.const import Const
+from mo.ops.reshape import Reshape
 from mo.ops.shape import Shape
 from mo.ops.strided_slice import StridedSlice
 
 
 class UnsqueezeTileReshapeBlockToInterpolate(MiddleReplacementPattern):
     """
     This transformation looks for Interpolation layer implemented using simple operations, i.e. Unsqueeze,
-    Tile, Reshape, and replaces found pattern with a sequence of Shape, StridedSlice, Const, Mul, Interpolate.
+    Tile, Reshape, and replaces found pattern with a sequence of Shape, StridedSlice, Mul, Interpolate.
 
-    Here we assume that the input of 'unsqueeze' is in NDHWC layout and is a 5D-tensor.
-
-    Found pattern will be replaced with
+    Found pattern
+        'something' -> Unsqueeze -> Tile -> Reshape
+    will be replaced with
         nodes=[
             ('shape', dict(kind='op', op='Shape')),
             ('strided_slice', dict(kind='op', op='StridedSlice')),
-            ('scales', dict(kind='op', op='Const')),
+            ('to_float', dict(kind='op', op='StridedSlice', dst_type=np.float32)),
+            ('m_scales', dict(kind='op', op='Const')),
             ('scaled_shape', dict(kind='op', op='Mul')),
+            ('floor', dict(kind='op', op='Floor')),
+            ('to_int', dict(kind='op', op='StridedSlice', dst_type=np.float32)),
+            ('scales', dict(kind='op', op='Const')),
+            ('axes', dict(kind='op', op='Const')),
             ('interp', dict(kind='op', op='Interpolate'))
         ],
         edges=[
+            ('something', 'interp', {'in': 0}),
+            ('something', 'shape', {'in': 0}),
             ('shape', 'strided_slice', {'in': 0}),
-            ('strided_slice', 'scaled_shape', {'in': 0}),
-            ('scales', 'scaled_shape', {'in': 1}),
-            ('scaled_shape', 'interp', {'in': 1}),
+            ('strided_slice', 'to_float', {'in': 0}),
+            ('to_float', 'scaled_shape', {'in': 0}),
+            ('m_scales', 'scaled_shape', {'in': 1}),
+            ('scaled_shape', 'floor'),
+            ('floor', 'to_int'),
+            ('to_int', 'interp', {'in': 1}),
+            ('scales', 'interp', {'in': 2}),
+            ('axes', 'interp', {'in': 3}),
         ]
```

Comment:
To be honest, I don't like current description of the transformation. Lets change the selected part to the following text:

```text
The transformation looks for a sub-graph performing unsqueeze-ing input tensor by some "axis" and then tiling over it fixed number of "times". This pattern can be represented with the Interpolate operation of mode "nearest" performing interpolation over specific "axis" with fixed output dimension size equal to "times". Note, that the transformation expects that the output from Tile is reshaped back to the tensor with rank equal to the input tensor rank. This constraints occurs because the pattern appears in the models where these patterns appear one after another, performing unsqueeze-ing over different dimensions, effectively performing interpolation over several dimensions. These sequences are merged in the `optimizer/extensions/middle/InterpolateSequenceToInterpolate.py` transformation into a single Interpolate operation.
```
And please, remove the `nodes, edges` list because they are hard to read and practically useless.

Reply:
Done.